README.md

```diff
@@ -16,6 +16,16 @@ The keys in the `env` secret are exposed in the `checkout` and `command` as envi
 
 ### AppRole Authentication
 
+## Environment Hook
```

Comment:
it would be good to have an explanation of `why` this section is in the readme - for example, why are we asking for users to set `VAULT_TOKEN` and `VAULT_SECRET_ID`. Maybe we also want to add a link to Vault's documentation on `AppRole Auth`?

Reply:
I think that's a good idea.